tests/ui/try-block/try-block-type-error.stderr

```diff
@@ -2,18 +2,13 @@ error[E0271]: type mismatch resolving `<Option<f32> as Try>::Output == {integer}
   --> $DIR/try-block-type-error.rs:10:9
    |
 LL |         42
-   |         ^^ expected `f32`, found integer
-   |
-help: use a float literal
-   |
-LL |         42.0
-   |           ++
+   |         ^^ expected integer, found `f32`
```

Comment:
This regresses :'(

Reply:
The message is right, it's just that the suggestion is gone, right?

Reply:
The message is not right ("expected integer" but pointing at an integer), because the way we are associating the obligation and the span is backwards. I don't think we should fix it though.

Reply:
Oh right yeah. Seems fine to break try blocks for now